### PR TITLE
Add mcxc242 counters support

### PR DESCRIPTION
**Pull request by michal-smola**
Enablement of NXP frdm_mcxc242 counters:
- Add counters configuration in soc devicetree.
- Add counters to board devicetree, update documentation and board yaml
- Enable acounter_basic_api test

### CODE REVIEW

File: tests/drivers/counter/counter_basic_api/boards/frdm_mcxc242.conf [hunk 1]
```diff
@@ -0,0 +1 @@
+CONFIG_SHARED_INTERRUPTS=y
```
Comment (danieldegrasse) [inline tests/drivers/counter/counter_basic_api/boards/frdm_mcxc242.conf:1]: I don't think we should be enabling this for this test only. Ideally, we should be selecting this Kconfig when `CONFIG_COUNTER_NXP_PIT` is enabled- but we don't need this for every platform. Probably the best way to drive this from the SOC defconfig, something like so:

```
if SOC_SERIES_MCXC # Change this to be more narrow if only MCXC242 SOC needs this enabled (IE if SOC_MCXC24)
config SHARED_INTERRUPTS
  default y if COUNTER_NXP_PIT
endif
```

Comment (michal-smola) [reply tests/drivers/counter/counter_basic_api/boards/frdm_mcxc242.conf:1]: @danieldegrasse, thank you for review. SHARED_INTERRUPTS is needed only if more than 1 PIT channel is enabled.  I haven't found how to select SHARED_INTERRUPTS based on count of enabled PIT channels.
Selecting it for 1 enabled channel increases flash footprint unnecessarily (MCXC has small flash), therefor I would prefer enabling it only if necessary.

Proposal: Enable only one PIT channel in default board dts. In the test, enable 2 channels and select SHARED_INTERRUPTS in counter_basic_api/boards/frdm_mcxc242.conf, with a comment why, to demonstrate the functionality.

Would you agree on the proposal or do you prefer to select SHARED_INTERRUPTS in SOC defconfig even for 1 enabled PIT channel?

File: dts/arm/nxp/nxp_mcxc_common.dtsi [hunk 2]
```diff
@@ -253,6 +253,52 @@
...
+				compatible = "nxp,pit-channel";
+				reg = <0>;
+				interrupts = <22 0>;
+				status = "disabled";
+			};
+
+			pit0_channel1: pit0_channel@1 {
+				compatible = "nxp,pit-channel";
+				reg = <1>;
+				interrupts = <22 0>;
```
Comment (decsny) [inline dts/arm/nxp/nxp_mcxc_common.dtsi:298]: you can put the interrupt on the parent pit node, see RT series DT for example, no need for config_shared_interrupts

**Review by decsny — CHANGES_REQUESTED** [review]

File: dts/arm/nxp/nxp_mcxc_common.dtsi (hunk 2 above)
Comment (michal-smola) [reply dts/arm/nxp/nxp_mcxc_common.dtsi]: Thank you. I have placed the interrupts to the parent node.

File: tests/drivers/counter/counter_basic_api/boards/frdm_mcxc242.conf (hunk 1 above)
Comment (michal-smola) [reply tests/drivers/counter/counter_basic_api/boards/frdm_mcxc242.conf]: I have placed the interrupts to the parent node as proposed by @decsny.

File: tests/drivers/counter/counter_basic_api/src/test_counter.c
Comment (danieldegrasse) [inline tests/drivers/counter/counter_basic_api/src/test_counter.c:1]: Commit message here should probably be updated- it seems that this commit is making a generic change for the MCUX RTC driver, that will affect more that just the MCXC242

Comment (michal-smola) [reply tests/drivers/counter/counter_basic_api/src/test_counter.c:1]: I have changed the commit message.

**Review by danieldegrasse — APPROVED** [review]

**Review by decsny — APPROVED** [review]
@EmilioCBen please review

**Comment by michal-smola** [comment]
@nordic-krch , could you review please?

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**